main.cpp

```diff
@@ -385,95 +315,162 @@ Step 3 is written below UDTs 5 - 9
+*/
+
+/*
+Thing 1) Concert
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) number of musicians (int)
+    2) the amount of ticket income (float)
+    3) number of audience (int)
+    4) the number of vendors around (int)
+    5) number of bands attend (int)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) perform many songs 
+    2) attract different audience
+    3) charge audience
  */
 
-/*
-Thing 6)
+/* Basketball Game
+Thing 2)
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) number of players (int)
+    2) number of coaching staff (int)
+    3) price of the ticket (double)
+    4) number of referee (int)
+    5) number of monitors (int)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) show different angle to audience
+    2) host the game
+    3) make profit
  */
 
 /*
-Thing 7)
+Thing 3) Product Manufacturing
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) number of production line (int)
+    2) number of operators (int)
+    3) number of engineers (int)
+    4) the amount of products made per week (int)
+    5) number of devices (int)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) assemble the products
+    2) solve engineering problems
+    3) increase the production rate
  */
 
 /*
-Thing 8)
+Thing 4) Guitar Tone Setting
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) the value of amp gain (float)
+    2) the number of guitar pedals (int)
+    3) the drive level of a overdrive pedal (float)
+    4) the number of reverb and delay pedals (int)
+    5) the number of different types of guitars (int)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) level up the guitar sound
+    2) make different sound effect
+    3) create more playing style
  */
 
 /*
-Thing 9)
-5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+Thing 5)
+5 properties: transmitter
+    1) hieght in m (int)
+    2) width in m (int)
+    3) distance in m (double)
+    4) amount of power consumed (float)
+    5) bandwidth (float)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) affect the data rate
+    2) increase battery usage
+    3) contain more frequency range
  */
 
 /*
-===================
-Part 1d assignment: Step 3
-===================
-You've just defined 10 UDTs!
-4 of them are unrelated (UDTs 1-4).
-5 of them form the sub-objects that make up the 10th UDT. 
-
-MOVE THEM BELOW this block comment and put them in numerical order 
-(1 - 10) starting on line 473
-    DO NOT COPY. 
-    I do not want to see 2 copies of your Plain-English UDTs.  
-    I only want to see the 10 UDTs written BELOW this block comment, in numerical order (1 - 10).
+Thing 6)
+5 properties: channel
+    1) number of obstacles (int)
+    2) number of unknown channels (int)
+    3) number of known channels (int)
+    4) distance in m (double)
+    5) density of medium (float)
+3 things it can do:
+    1) reflect the signal
+    2) deflect the signal
+    3) change the transmit speed
```

Comment:
This class is supposed to be one of the properties of your Communications System.
What you have described via these properties is not a channel, but something else.
A communication system channel would have properties like: 
- bitrate
- port number
- block size
- number of packets dropped.
- channel address, such as the IP address.

What you have described seems to be referring to a physical communications cable, like a fiber optic cable or ethernet cable.

Revise this Thing. 
choose properties that directly relate to the Thing you choose.
- [ ] revise